sbi/analysis/plot.py

```diff
@@ -976,3 +978,353 @@ def _get_default_opts():
         },
         "title_format": {"fontsize": 16},
     }
+
+
+def sbc_rank_plot(
+    ranks: Union[Tensor, np.ndarray, List[Tensor] , List[np.ndarray]],
+    num_posterior_samples: int,
+    num_bins: int = None,
+    num_repeats: int = 50,
+    plot_type: str = "cdf",
+    parameter_labels: List[str] = None,
+    ranks_labels: List[str] = None,
+    colors: List[str] = None,
+    line_alpha: float = 0.8,
+    show_uniform_region: bool = True,
+    uniform_region_alpha: float = 0.3,
+    xlim_offset_factor: float = 0.1,
+    num_cols: int = 4,
+    params_in_subplots: bool = False,
+    show_ylabel: bool = False,
+    sharey: bool = False,
+    fig: Figure = None,
+    ax: Axes = None,
+    figsize: tuple = None,
+):
+    """Plot simulation-based calibration ranks as empirical CDFs.
+
+    Args:
+        ranks: Tensor of ranks to be plotted shape (num_sbc_runs, num_parameters), or
+            list of Tensors when comparing several sets of ranks, e.g., set of ranks
+            obtained from different methods.
+        num_bins: number of bins used for binning the ranks, default is
+            num_sbc_runs / 20.
+        num_repeats: number of repeats for each empirical CDF step (resolution).
+        plot_type: type of SBC plot, histograms ("hist") or empirical cdfs ("cdf").
+        parameter_labels: list of labels for each parameter dimension.
+        ranks_labels: list of labels for each set of ranks.
+        colors: list of colors for each parameter dimension, or each set of ranks.
+        line_alpha: alpha for cdf lines or histograms.
+        show_uniform_region: whether to plot the region showing the cdfs expected under
+            uniformity.
+        uniform_region_alpha: alpha for region showing the cdfs expected under
+            uniformity.
+        xlim_offset_factor: factor for empty space left and right of the histogram.
+        num_cols: number of subplot columns, e.g., when plotting ranks of many
+            parameters.
+        params_in_subplots: whether to show each parameter in a separate subplot, or
+            all in one.
+        fig: figure object to plot in.
+        ax: axis object, must have shape (number of set of ranks,) or be a raw axes
+            object.
+        figsize: dimensions of figure object, defaults to (8, 5) or (number of set of
+            ranks * 4, 5).
+
+    Returns:
+        fig, ax: figure and axis objects.
+
+    """
+
+    if isinstance(ranks, (Tensor, np.ndarray)):
+        ranks = [ranks]
```

Comment:
this will give a pyright error. You can not redeclare the an existing variable to a new type. You have to name the new variable (which has a different type) differently, e.g. `ranks_list`.